webpack/assets/javascripts/react_app/components/Layout/NavigationSearch.js

```diff
@@ -0,0 +1,217 @@
+      } else {
+        navLinks[group.title] = {
+          ...group,
+          parents: [item.title, parent].filter(Boolean),
+        };
+      }
+    });
+    parent = null;
+  });
+
+  const navItems = Object.keys(navLinks);
+  const menuNav = navItem => (
+    <MenuItem
+      to={navLinks[navItem].href}
+      onClick={event =>
+        clickAndNavigate(
+          navLinks[navItem].onClick,
+          navLinks[navItem].href,
+          event
+        )
+      }
+      itemId={navItem}
+      key={navItem}
+      description={[...navLinks[navItem].parents, navItem].join(' > ')}
+    >
+      {navItem}
+    </MenuItem>
+  );
+  const [autocompleteOptions, setAutocompleteOptions] = useState(
+    navItems.slice(0, 10).map(menuNav)
+  );
+  const [value, setValue] = useState('');
+  const [hint, setHint] = useState('');
+
+  const [isAutocompleteOpen, setIsAutocompleteOpen] = useState(false);
+
+  const searchInputRef = useRef(null);
+  const autocompleteRef = useRef(null);
+
+  const onClear = () => {
+    setValue('');
+  };
+
+  const onChange = newValue => {
+    if (
+      newValue !== '' &&
+      searchInputRef?.current?.contains(document.activeElement)
+    ) {
+      setIsAutocompleteOpen(true);
+
+      // When the value of the search input changes, build a list of no more than 10 autocomplete options.
+      // Options which start with the search input value are listed first, followed by options which contain
+      // the search input value.
+      let options = navItems
+        .filter(option => option.toLowerCase().includes(newValue.toLowerCase()))
+        .map(menuNav);
+      if (options.length > 10) {
+        options = options.slice(0, 10);
+      }
+
+      // The menu is hidden if there are no options
+      setIsAutocompleteOpen(options.length > 0);
+
+      setAutocompleteOptions(options);
+    } else {
+      setIsAutocompleteOpen(false);
+    }
+    setValue(newValue);
+  };
+  // Whenever an autocomplete option is selected, set the search input value, close the menu, and put the browser
+  // focus back on the search input
+  const onSelect = e => {
+    e.stopPropagation();
+    setIsAutocompleteOpen(false);
+  };
+
+  React.useEffect(() => {
+    const handleClick = event => {
+      if (
+        isAutocompleteOpen &&
+        autocompleteRef?.current &&
+        !autocompleteRef.current.contains(event.target)
+      ) {
+        // The autocomplete menu should close if the user clicks outside the menu.
+        setIsAutocompleteOpen(false);
+      } else if (
+        !isAutocompleteOpen &&
+        searchInputRef?.current &&
+        searchInputRef.current.contains(event.target)
+      ) {
+        // The autocomplete menu should open if the user clicks on the search input.
+        setIsAutocompleteOpen(true);
+      }
+    };
+    const handleMenuKeys = event => {
+      // If there is a hint while the browser focus is on the search input, tab or right arrow will 'accept' the hint value
```

Comment:
That doesn't seem to be working... Also, I don't see where is `setHint` being used except reseting below.

Reply:
leftovers from copying PF logic